session/auth/authprovider/tokenseed.go

```diff
@@ -0,0 +1,65 @@
+package authprovider
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+type tokenSeeds struct {
+	mu  sync.Mutex
+	dir string
+}
+
+type seed struct {
+	Seed []byte
+}
+
+func (ts *tokenSeeds) getSeed(host string) ([]byte, error) {
+	ts.mu.Lock()
+	defer ts.mu.Unlock()
+
+	fp := filepath.Join(ts.dir, ".token_seed")
```

Comment:
Was hoping to avoid this but couldn't think of a better solution. Afaics if we don't do something like this we always have a possibility for a chosen-plaintext attack from the daemon side. Another possible way would be to use pbkdf/scrypt for the verification but didn't want to take the 100ms+ performance hit for this and waste cpu.

Reply:
Worth leaving a comment that describes this?

Reply:
done

Reply:
fyi @justincormack